### PR TITLE
media upload functionality

### PR DESCRIPTION
**Pull request by ALI-KHALED-13**

### CODE REVIEW

File: api/utils/handleMedia/index.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    Bucket: process.env.BUCKET_NAME,
+    Key: fileKey
+  });
+
+  const url = await getSignedUrl(s3, getCommand, {expiresIn: 604800}); // 7 days
+
+  return url;
+}
+
+const populatePFPUrl = async (user)=> {
```
Comment (Elena-Erganyan) [inline api/utils/handleMedia/index.js:46]: When I tried to login with a newly registered user, I got this error : `internal server Error: No value provided for input HTTP label: Key`.
It seems that `user.pfp` is an empty object, so it runs `createSignedUrl` and can't find `fileKey`.

File: api/products/controllers.js [hunk 2]
```diff
@@ -1,13 +1,18 @@
 const Product = require("../../models/Product");
 const User = require("../../models/User");
 const { DB404Error, UnauthorizedError } = require("../utils/catchErrors");
+const { createSignedUrl, populatePFPUrl } = require("../utils/handleMedia");
```
Comment (Elena-Erganyan) [inline api/products/controllers.js:4]: Unused `createSignedUrl`

**Review by Elena-Erganyan — CHANGES_REQUESTED** [review]
Found just one issue.
Other than that everything works fine, good job :)

File: api/utils/handleMedia/index.js (hunk 1 above)
Comment (ALI-KHALED-13) [reply api/utils/handleMedia/index.js:46]: thanks for figuring out the reason, will fix it

File: package.json [hunk 3]
```diff
@@ -4,13 +4,17 @@
   "description": "",
   "main": "server.js",
   "scripts": {
-    "test": "jest --verbose"
+    "test": "jest --verbose",
+    "server": "nodemon server",
+    "client": "npm run dev --prefix client",
+    "app": "concurrently \"npm run dev\" \"npm run client\""
```
Comment (Elena-Erganyan) [inline package.json:10]: 1) Better to add `concurrently` in `devDependencies`
2) It should be `npm run server` instead of `npm run dev`

**Review by Elena-Erganyan — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (Elena-Erganyan) [reply package.json:10]: Hmm, regarding the 1., `nodemon` should be there too actually 😅

**Pull request merged**